base/array.jl

```diff
@@ -1806,15 +1806,22 @@ julia> filter(isodd, a)
 """
 filter(f, As::AbstractArray) = As[map(f, As)::AbstractArray{Bool}]
 
-function filter!(f, a::Vector)
-    insrt = 1
-    for acurr in a
-        if f(acurr)
-            a[insrt] = acurr
-            insrt += 1
+function filter!(f, a::AbstractVector)
+    if !isempty(a)
```

Comment:
I get not wanting to wrap the entire thing in an `if`, but I generally try to avoid putting multiple return statements in a function.